src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1555,12 +1555,50 @@ RebalanceTableShards(RebalanceOptions *options, Oid shardReplicationModeOid)
 		return;
 	}
 
+	/* find the name of the shard transfer mode to interpolate in the scheduled command */
+	Datum shardTranferModeLabelDatum =
+		DirectFunctionCall1(enum_out, shardReplicationModeOid);
+	char *shardTranferModeLabel = DatumGetCString(shardTranferModeLabelDatum);
+
+	if (list_length(placementUpdateList) == 0)
+	{
+		return;
+	}
+
+	/* schedule planned moves */
+	int64 jobId = CreateBackgroundJob("rebalance", "Rebalance colocation group ...");
+
+	PlacementUpdateEvent *move = NULL;
+	StringInfoData buf = { 0 };
+	initStringInfo(&buf);
+	bool first = true;
+	int64 prevJobId = 0;
+	foreach_ptr(move, placementUpdateList)
+	{
+		resetStringInfo(&buf);
+
+		appendStringInfo(&buf,
+						 "SELECT citus_move_shard_placement(%ld,%s,%u,%s,%u,%s)",
```

Comment:
By running this in the background we get a UX decrease. Any pre-checks that citus_move_shard_placement does now only run in the background. e.g. I now get this error in the tasks table, instead of when running rebalance_table_shards:
```
message     │ ERROR: cannot use logical replication to transfer shards of the relation t since it doesn't have a REPLICA IDENTITY or PRIMARY KEY          ↵
            │ DETAIL: UPDATE and DELETE commands on the shard will error out during logical replication unless there is a REPLICA IDENTITY or PRIMARY KEY.↵
            │ HINT: If you wish to continue without a replica identity set the shard_transfer_mode to 'force_logical' or 'block_writes'.                  ↵
```

It would be good to do these pre-checks before creating the tasks.

Reply:
In the same idea, I think we want to push the ensure reference tables exists portion of the rebalance as a first step in the tasks to perform, if required - although we can simply _always_ start with it.